### ISSUE TITLE
[FEATURE REQUEST] Automatically delete old cookies that were grey/whitelisted

### ISSUE THREAD

**Issue by ghost** [mrdokenny/Cookie-AutoDelete#707]
So I was looking for an extension that automatically deletes old cookies. For example, if I set a "delay" of 30 days, it should delete all cookies that are 30 days old (or older) automatically, based on the mandatory "Created" timestamp.
**Is your feature request related to a problem? Please describe.**
A delay of 2,592,000 seconds (= 30 days) in CADs settings (delay after tab closed) apparently is too high and it just deletes the cookies immediately.
**Describe the solution you'd like**
Since I have a password manager, signing back into websites every 30 days isn't a hassle, but after every tab or window close seems like a little too much for me. Since there are no other good automatic cookie manager extensions for Firefox out there, I'd be very grateful if this could be implemented here.
**Implementation thoughts**
Looking at the source code, an interval could trigger the requested function every 10 minutes (this interval could also be set in the settings) or so in the src/background.ts file, which checks for outdated cookies based on an "expiry time" from the settings.

**Comment by kennethtran93** [comment]
Normally cookies that are expired should be cleared by the browser itself.

As for the interval removal, it would just be another trigger to the existing cleanup process that CAD does.  That is already requested for.

Because we only see the expiry time, we don't know when the cookie was created, unless we store it ourselves.

In addition, per https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/getAll, we do not get any expired cookies.

**Comment by kennethtran93** [comment]
You've also raised a bug due to setTimeout using a 32 bit int to store the delay so the max value allowed would be `2,147,483,647`

What you had specified was 2,592,000 seconds x 1000 = 2,592,000,000 milliseconds which is over what the 32-bit.  I'll need to update the code to prevent going over the max number.

Source: https://stackoverflow.com/questions/3468607/why-does-settimeout-break-for-large-millisecond-delay-values

**Comment by ghost** [comment]
> Normally cookies that are expired should be cleared by the browser itself.

Correct. I’m asking to implement an automatic deletion after `n` days have passed since the Cookie was created, regardless of its expiry date (unless it was deleted earlier, of course).

> As for the interval removal, it would just be another trigger to the existing cleanup process that CAD does. That is already requested for.

Great

> Because we only see the expiry time, we don't know when the cookie was created, unless we store it ourselves.

I suggest to create a cookie for every cookie added containing the creation timestamp and a reference to the actual cookie (could also be stored via the storage api). That data could then be used to track and delete cookies after a certain amount of time, see https://stackoverflow.com/a/39912808 .

**Comment by kennethtran93** [comment]
> Correct. I’m asking to implement an automatic deletion after `n` days have passed since the Cookie was created, regardless of its expiry date (unless it was deleted earlier, of course).

Will this be for a global 'all' cookies regardless of domain or another option within each expression/container?  The former seems simple to implement but not the latter.

> I suggest to create a cookie for every cookie added containing the creation timestamp and a reference to the actual cookie (could also be stored via the storage api). That data could then be used to track and delete cookies after a certain amount of time, see https://stackoverflow.com/a/39912808 .

I suppose going by storage seems better, otherwise the cookie count will need to be filtered out accordingly.  Only local storage though - I don't plan on making the cookie expiration stuff in sync as that would take up too much space depending on the amount of cookies to keep track of...

In any case, it would be on the list of lower priorities at this time.

**Comment by phoglenix** [comment]
Firefox has a pref network.cookie.maxageCap that seems to implement the desired behaviour here. But I'm not sure if it clears other forms of site data besides cookies (likely not).

It would still be great to have this supported by an extension, for other browsers and more reliability than an undocumented pref value.
